File: ms1-core-api/src/config/env.ts
import dotenv from 'dotenv';

dotenv.config();

const requiredVars = ['PORT', 'DATABASE_URL', 'NODE_ENV'];

for (const key of requiredVars) {
  if (!process.env[key]) {
    throw new Error(`Missing required environment variable: ${key}`);
  }
}

export const env = {
  PORT: parseInt(process.env.PORT as string, 10),
  DATABASE_URL: process.env.DATABASE_URL as string,
  JWT_SECRET: process.env.JWT_SECRET as string,
  NODE_ENV: process.env.NODE_ENV as string,
  REDIS_URL: process.env.REDIS_URL || 'redis://localhost:6379',
  MS2_BASE_URL: process.env.MS2_BASE_URL || 'http://localhost:7000',
  // Shared secret used to authenticate webhook calls from MS2.
  // If set, MS1 will validate the X-Webhook-Secret header on incoming webhooks.
  MS2_WEBHOOK_SECRET: process.env.MS2_WEBHOOK_SECRET || '',
};

File: ms2-agent/app/config/env.py
import os
from dotenv import load_dotenv

load_dotenv()

_required = [
    "FASTAPI_PORT",
    "MS2_DATABASE_URL",
    "NEO4J_URI",
    "NEO4J_USERNAME",
    "NEO4J_PASSWORD",
    "PYTHON_ENV",
    "MS1_BASE_URL",
]

for _key in _required:
    if not os.getenv(_key):
        raise RuntimeError(f"Missing required environment variable: {_key}")


class Env:
    FASTAPI_PORT: int = int(os.environ["FASTAPI_PORT"])
    # MS2's own PostgreSQL database — NEVER the same as MS1's DATABASE_URL.
    MS2_DATABASE_URL: str = os.environ["MS2_DATABASE_URL"]
    NEO4J_URI: str = os.environ["NEO4J_URI"]
    NEO4J_USERNAME: str = os.environ["NEO4J_USERNAME"]
    NEO4J_PASSWORD: str = os.environ["NEO4J_PASSWORD"]
    PYTHON_ENV: str = os.environ["PYTHON_ENV"]
    # Root directory where per-analysis workspace folders are created.
    # Defaults to ./workspace relative to the working directory.
    WORKSPACE_ROOT: str = os.environ.get("WORKSPACE_ROOT", "./workspace")
    # MS1 base URL — used when sending webhook callbacks from MS2 to MS1.
    MS1_BASE_URL: str = os.environ["MS1_BASE_URL"]
    # Shared secret for authenticating webhook calls to MS1.
    # Must match MS2_WEBHOOK_SECRET in ms1-core-api/.env.
    MS2_WEBHOOK_SECRET: str = os.environ.get("MS2_WEBHOOK_SECRET", "")


env = Env()
